### PR TITLE
Implemented API Host Collection CRUDs

### PR DESCRIPTION
**Pull request by abalakh**
Test results:
```python
py.test -v tests/foreman/api/test_hostcollection.py
================================ test session starts =================================
platform linux2 -- Python 2.7.10, pytest-2.8.2, py-1.4.30, pluggy-0.3.1 -- /home/andrii/env/bin/python
cachedir: .cache
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1
collected 17 items 

tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_negative_create_with_invalid_name PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_add_system PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_add_systems PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_create_with_description PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_create_with_limit PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_create_with_name PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_create_with_system PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_create_with_systems PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_create_with_unlimited_content_hosts PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_delete PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_read_system_ids <- robottelo/decorators.py SKIPPED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_update_description PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_update_limit PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_update_name PASSED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_update_system <- robottelo/decorators.py SKIPPED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_update_systems <- robottelo/decorators.py SKIPPED
tests/foreman/api/test_hostcollection.py::HostCollectionTestCase::test_positive_update_unlimited_content_hosts PASSED

======================= 14 passed, 3 skipped in 93.89 seconds ========================
```

### CODE REVIEW

**Comment by abalakh** [comment]
Depends on SatelliteQE/nailgun#249

**Comment by oshtaier** [comment]
ACK

File: tests/foreman/api/test_hostcollection.py [hunk 1]
```diff
@@ -19,10 +22,80 @@ def setUpClass(cls):
...
+            with self.subTest(unlimited):
+                host_collection = entities.HostCollection(
+                    organization=self.org,
+                    unlimited_content_hosts=unlimited,
+                ).create()
+                self.assertEqual(
+                    host_collection.unlimited_content_hosts, unlimited)
+
+    @tier1
     def test_positive_create_with_system(self):
```
Comment (sghai) [inline tests/foreman/api/test_hostcollection.py]: I think we should use replace `system` with `content_host`. This applies to all occurrences of system.

**Comment by oshtaier** [comment]
+1 to @sghai  comment

**Comment by sghai** [comment]
ACK.. Changes looks great. 

Only thing is we are using `system` keyword everywhere in test name though test description correctly using `Content Host`. 

Its a minor and non-blocking comment since our api's too using system ;)

**Comment by abalakh** [comment]
Sure, will update. Wasn't sure about `system`/`content_host` as api uses `system` parameter for host collections, nailgun uses `System` entity etc.

**Comment by sghai** [comment]
Yeah.. this all will be unified and related stuff is still under development in upstream. Going forward `system` will be completely replaced by `host` entity.

**Comment by abalakh** [comment]
@sghai updated, thanks for clarification.

File: tests/foreman/api/test_hostcollection.py [hunk 2]
```diff
@@ -19,10 +22,80 @@ def setUpClass(cls):
...
+        for limit in (1, 3, 5, 10, 20):
+            with self.subTest(limit):
+                host_collection = entities.HostCollection(
+                    max_content_hosts=limit,
+                    organization=self.org,
+                ).create()
+                self.assertEqual(host_collection.max_content_hosts, limit)
+
+    @tier1
+    def test_positive_create_with_unlimited_content_hosts(self):
```
Comment (sghai) [inline tests/foreman/api/test_hostcollection.py]: `test_positive_create_with_unlimited_chosts` ?

File: tests/foreman/api/test_hostcollection.py [hunk 3]
```diff
@@ -66,3 +169,149 @@ def test_positive_read_system_ids(self):
...
+            unlimited_content_hosts=False,
+        ).create()
+        for limit in (1, 3, 5, 10, 20):
+            with self.subTest(limit):
+                host_collection.max_content_hosts = limit
+                self.assertEqual(
+                    host_collection.update().max_content_hosts, limit)
+
+    @tier1
+    def test_positive_update_unlimited_content_hosts(self):
```
Comment (sghai) [inline tests/foreman/api/test_hostcollection.py]: `test_positive_update_unlimited_chosts` ?

Comment (sghai) [inline tests/foreman/api/test_hostcollection.py]: `test_positive_update_with_unlimited_chosts`

**Comment by sghai** [comment]
@abalakh : I found two places where we were using content_host instead of chost. Could you please update this for consistency sake. I'll merge the PR as soon as you fix this. Thank you for the changes.

File: tests/foreman/api/test_hostcollection.py (hunk 3 above)
Comment (sghai) [inline tests/foreman/api/test_hostcollection.py]: Clarified on IRC, the param name is `unlimited_content_hosts` so the test name like this. Thanks @abalakh

**Comment by sghai** [comment]
ACK and merge.

**Comment by sthirugn** [comment]
I think `test_positive_update_with_unlimited_chosts` is better, we do not have rules around keeping the parameter name as it is in the test case name.  I vote for `chost` wherever `content_host` is used in test cases.

**Comment by sthirugn** [comment]
My comment is addressed in https://github.com/SatelliteQE/robottelo/pull/3189

**Pull request merged**